### ISSUE TITLE
FB9024329: The `.font` environment key should always be defined in the environment

### ISSUE THREAD

**Issue by sindresorhus** [feedback-assistant/reports#195]
- **Date:** 2021-03-01
- **Resolution:** Open
- **Area:** SwiftUI Framework
- **OS:** macOS 11.2.1
- **Type:** Suggestion

## Description

Currently, it’s only defined if I explicitly set `.font(…)` in a parent view. However, I would argue it should always be defined because the rendered output always has a size, it’s just inferred. Having the font always be defined would enable use-cases like creating modifiers that work on the inferred font.

For example, I wanted to make a modifier that lets me make any `Text` use monospaced digits without knowing the existing font, font size, or font weight:

```swift
private struct MonospacedDigit: ViewModifier {
	func body(content: Content) -> some View {
		content.transformEnvironment(\.font) {
			$0 = $0?.monospacedDigit()
		}
	}
}

extension View {
	/**
	Adjusts the existing font to use monospace digits.

	```
	Text("Foo 10:20")
		.monospacedDigit()
	```
	*/
	func monospacedDigit() -> some View {
		modifier(MonospacedDigit())
	}
}
```

The problem with this one is that it only works if I explicitly set `.font()` which I often don’t do as usually the default font is fine.

I could maybe make it default to the system default font if the font is not defined in the environment, but it’s unclear to me whether that is correct. For example, the default inferred font size might differ depending on where the SwiftUI view is used. For example, whether it’s used directly in a `HStack` or as a `Button` label where the `Button` has `.controlSize(.small)`.

In short, there’s a lot of useful font-related modifiers that could be made if the `.font` was always defined to be the inferred font.
